### ISSUE TITLE
XSDL for UserLAnd

### ISSUE THREAD

**Issue by luongthomas** [pelya/xserver-xsdl#97]
Hi Pelya,

I'm on the UserLAnd development team working along side with Corbin.  Currently I've been tasked with integrating XSDL support in UserLAnd and it's coming along well thanks to the info you've provided [here](https://github.com/corbinlc/GNURootDebian/issues/37).   

On a separate note, our test devices on Android 9.0 can reproduce the same behavior as issue #62.  Is there any way we can lend a hand towards solving this issue?

**Comment by pelya** [comment]
The only advice I can give you is to debug XSDL step by step. You'll need
to use ancient Android SDK and NDK and roll back SDL repo to some old
version to even compile XSDL. I was trying to update my build scripts, but
it still won't work with new NDK.

**Comment by luongthomas** [comment]
Will you be planning on spending more time debugging/maintaining XSDL going forward?

**Comment by pelya** [comment]
Yes, I'm planning to update it, but my current development speed is
measured in years, not weeks or months. I'll merge your patches though.

First planned task is to make it compile with NDK r19.

**Comment by corbinlc** [comment]
We are going to go ahead and add support for sending an intent to XSDL now, but we have version guarding to only allow it if your SDK version shows that you are less than android 9.  Is it believed that by using a new NDK, that the current problem would be resolved?

**Comment by pelya** [comment]
Yes, there's no need to limit XSDL to Android 2.3 anymore, since Google
removed support for Android 4.0 and older from NDK.

**Comment by MathiasMagnus** [comment]
I was messing around with UserLAnd and XSDL, because I was trying to get Visual Studio Code working on my Xiaomi Mi Max 3 handset (7" phablet running Android Pie). I managed to install an arm64 build of VS Code, but I can't launch it, because many X parts are missing, due to not being able to launch an X Server. @luongthomas should UserLAnd integrate XSDL, that would simplify the life of tinkerers, such as myself. Making ends meet any other way is also welcome.

**Comment by pelya** [comment]
1.20.41 fixed Android 9 support, and updated to latest NDK.

**Comment by corbinlc** [comment]
@pelya, I will test on a Android 9 device later, but it is failing on my Android 10 device.

**Comment by JanuszChmiel** [comment]
I can confirm, that this Xserver XSDL release, which is now available on Google Play is really revolutionar. Pulseaudio really works at The background, its lateny is much more smaller, app is much more stable than before.
I Am only heavily fighting with Audacity on Userland when running it from Debian.
Audacity shows debug error message related to top window.
It will always run from Termux and from Debian which have been prepared by special script for Termux.
I have even used radio player which streamed live radio stream and I could play it at The background with no issues at all.
I have only Android 6.0 device. But I AM very very satisfied with Pelia work. 
There is The only one big issue when running Userland with more complex desktop environments. Many of them is based on functioning Udev for detecting hod keys. So for example, LXDE can not detect pressed hodkeys such as ALT+F1. So it is much more simple to simply run standalone app or more of them than triing to run whole individual desktop environment, which is based on Udev for detecting hod keys.
It is sad, that there is no possibility to emulate Udev partially to support hod keys detection with no rooting devices.

**Comment by NuKESec** [comment]
I'll be uninstalling UserLand. there has been talk of xsdl support since android 9.0 was released, and still nonexistent with no progress at all. the app is worthless without xsdl support.

**Comment by pelya** [comment]
@corbinlc I've tried XSDL on Android 10 emulator (x86_64), and it works okay. Could you please copy logcat from your device where it fails?
